### PR TITLE
Add mask input topic

### LINKED ISSUE #352: Mask component documentation
Create topic for Mask input web component.

### PR DESCRIPTION
**Pull request by onlyexeption**
Closes #352

### CODE REVIEW

File: doc/en/components/inputs/mask-input.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+Applying the mask is pretty straightforward. All you need to do is provide a predetermined pattern to the `mask` property of the input.
+
+In the example below, we will apply a mask for a phone number with an extension code.
+
+```html
+<igc-mask-input id="mask-input" mask="(####) 00-00-00 Ext. 9999">
+    <igc-icon name="github" slot="prefix"></igc-icon>
+    <igc-icon name="github" slot="suffix"></igc-icon>
+    <span slot="helper-text">This is some helper text</span>
```
Comment (igdmdimitrov) [inline doc/en/components/inputs/mask-input.md:65]: Change helper text to something related to the sample, for example "Phone number" or something else.

File: doc/en/components/inputs/mask-input.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+### Applying Mask
+
+Applying the mask is pretty straightforward. All you need to do is provide a predetermined pattern to the `mask` property of the input.
+
+In the example below, we will apply a mask for a phone number with an extension code.
+
+```html
+<igc-mask-input id="mask-input" mask="(####) 00-00-00 Ext. 9999">
+    <igc-icon name="github" slot="prefix"></igc-icon>
+    <igc-icon name="github" slot="suffix"></igc-icon>
```
Comment (igdmdimitrov) [inline doc/en/components/inputs/mask-input.md:64]: Change the prefix icon to some kind of phone. You can delete the suffix.

File: doc/en/components/inputs/mask-input.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+| 0 | requires a digit (0-9) |
+| 9 | requires a digit (0-9) or a space |
+| # | requires a digit (0-9), plus (+), or minus (-) sign |
+| L | requires a letter (a-Z) |
+| ? | requires a letter (a-Z) or a space |
+| A | requires an alphanumeric (0-9, a-Z) |
+| a | requires an alphanumeric (0-9, a-Z) or a space |
+| & | any keyboard character (excluding space) |
+| C | any keyboard character |
+
```
Comment (rkaraivanov) [inline doc/en/components/inputs/mask-input.md:54]: | Mask Character | Description |
| :--- | :--- |
| \ | Escapes a mask flag and turns it into a literal. |

Comment (rkaraivanov) [reply doc/en/components/inputs/mask-input.md:54]: Also, there should be a mention about required flags and validation.
Refer to [this](https://github.com/IgniteUI/igniteui-webcomponents/wiki/Masked-input#functionality) section

**Review by rkaraivanov — CHANGES_REQUESTED** [review]
Except the comments above, LGTM

File: doc/en/components/inputs/mask-input.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+| # | requires a digit (0-9), plus (+), or minus (-) sign |
+| L | requires a letter (a-Z) |
+| ? | requires a letter (a-Z) or a space |
+| A | requires an alphanumeric (0-9, a-Z) |
+| a | requires an alphanumeric (0-9, a-Z) or a space |
+| & | any keyboard character (excluding space) |
+| C | any keyboard character |
+| \ | escapes a mask flag and turns it into a literal |
+
+These flags also participate in the component validation - i.e., the input becomes invalid if some, but not all required positions are filled (no positions filled/empty value is still a responsibility of `required`). This applies to both stand-alone inputs and when included in a form.
```
Comment (igdmdimitrov) [inline doc/en/components/inputs/mask-input.md:56]:
```suggestion
These flags also participate in the component validation - i.e., the input becomes invalid if some but not all required positions are filled (no positions filled/empty value is still a responsibility of `required`). This applies to both stand-alone inputs and when included in a form.
```

**Review by igdmdimitrov — APPROVED** [review]

File: doc/en/components/inputs/mask-input.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+| :--- | :--- |
+| 0 | requires a digit (0-9) |
+| 9 | requires a digit (0-9) or a space |
+| # | requires a digit (0-9), plus (+), or minus (-) sign |
+| L | requires a letter (a-Z) |
+| ? | requires a letter (a-Z) or a space |
+| A | requires an alphanumeric (0-9, a-Z) |
+| a | requires an alphanumeric (0-9, a-Z) or a space |
+| & | any keyboard character (excluding space) |
+| C | any keyboard character |
```
Comment (rkaraivanov) [inline doc/en/components/inputs/mask-input.md:53]: I suggest using the table from the specification. Currently there is no distinction about which mask flags are required and which are optional.

**Review by rkaraivanov — CHANGES_REQUESTED** [review]

**Review by HUSSAR-mtrela — APPROVED** [review]

**Review by HUSSAR-mtrela — APPROVED** [review]

**Pull request merged**